mlflow/tensorflow.py

```diff
@@ -1015,17 +1015,22 @@ def _get_early_stop_callback_attrs(callback):
         except Exception:  # pylint: disable=W0703
             return None
 
-    def _log_early_stop_callback_metrics(callback, history, metrics_logger):
+    def _log_early_stop_callback_metrics(callback, history, expected_last_epoch, metrics_logger):
         if callback:
             callback_attrs = _get_early_stop_callback_attrs(callback)
             if callback_attrs is None:
                 return
             stopped_epoch, restore_best_weights, patience = callback_attrs
             metrics_logger.record_metrics({"stopped_epoch": stopped_epoch})
 
-            # Weights are restored only if early stopping occurs
-            if stopped_epoch != 0 and restore_best_weights:
-                restored_epoch = stopped_epoch - max(1, patience)
```

Comment:
If patience is zero, then `restored_epoch` should be equivalent to `stopped_epoch`. This `max()` logic does not appear to be correct.